WcaOnRails/spec/models/competition_spec.rb

```diff
@@ -23,6 +23,23 @@
     end
   end
 
+  context "delegates" do
+    it "delegates for future comps must be delegates" do
+      competition = FactoryGirl.build :competition, :with_delegate, starts: 1.year.from_now
+      competition.delegates.first.update_columns(delegate_status: nil)
+
+      expect(competition).to be_invalid
+      expect(competition.errors.messages[:delegate_ids]).to eq ["are not all delegates"]
+    end
+
+    it "delegates for past comps may no longer be delegates" do
```

Comment:
This has an unfortunate double meaning ("you may not walk on the grass"). I'd replace it with "need".